moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -807,10 +807,17 @@ bool ServoCalcs::enforcePositionLimits(trajectory_msgs::msg::JointTrajectory& jo
             std::find(joint_trajectory.joint_names.begin(), joint_trajectory.joint_names.end(), joint->getName());
         auto joint_idx = std::distance(joint_trajectory.joint_names.begin(), joint_itr);
 
-        if ((joint_trajectory.points[0].velocities[joint_idx] < 0 &&
-             (joint_angle < (limits[0].min_position + parameters_->joint_limit_margin))) ||
-            (joint_trajectory.points[0].velocities[joint_idx] > 0 &&
-             (joint_angle > (limits[0].max_position - parameters_->joint_limit_margin))))
+        const bool near_min_position =
+            parameters_->publish_joint_velocities ?
+                joint_trajectory.points[0].velocities[joint_idx] < 0 &&
```

Comment:
If you change all the `operator[]` to `at()` it should help catch any future occurrences of this too.